app/services/change_schedule.rb

```diff
@@ -133,6 +132,12 @@ def validate_new_schedule_valid_with_existing_declarations
     end
   end
 
+  def applicable_declarations
+    @applicable_declarations ||= participant_profile.participant_declarations.select do |declaration|
+      declaration.state.in?(%w[submitted eligible payable paid])
+    end
```

Comment:
We should do a `participant_declarations.where(state: %w[submitted eligible payable paid])` here to avoid loading all the declarations unnecessarily.